### PR TITLE
Added shogi model and agent

### PR DESCRIPTION
**Pull request by LarsKluijtmans**

### CODE REVIEW

File: functions/services/ai/environment.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        ]
+        piece = piece_types[piece_type - 1]
+        return piece
+
+    def get_legal_moves(self) -> list[Move]:
+        """Get all legal moves"""
+        generator = self.board.generate_legal_moves()
+        legal_moves = [move for move in generator if move.from_square is not None]
+        if len(legal_moves) == 0:
+            raise NoMovesException()
```
Comment (airellbachtiar) [inline functions/services/ai/environment.py]: return []

File: functions/services/ai/exceptions/no_legal_moves.py
Comment (airellbachtiar) [inline functions/services/ai/exceptions/no_legal_moves.py:1]: this can be removed

File: functions/services/ai/environment.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            "PROM_BISHOP",
+            "PROM_ROOK",
+        ]
+        piece = piece_types[piece_type - 1]
+        return piece
+
+    def get_legal_moves(self) -> list[Move]:
+        """Get all legal moves"""
+        generator = self.board.generate_legal_moves()
+        legal_moves = [move for move in generator if move.from_square is not None]
```
Comment (airellbachtiar) [inline functions/services/ai/environment.py]: [move for move in generator]

File: functions/services/ai/environment.py (hunk 1 above)
Comment (LarsKluijtmans) [reply functions/services/ai/environment.py]: Done!

File: functions/services/ai/environment.py (hunk 2 above)
Comment (LarsKluijtmans) [reply functions/services/ai/environment.py]: Done

File: functions/services/ai/exceptions/no_legal_moves.py
Comment (LarsKluijtmans) [reply functions/services/ai/exceptions/no_legal_moves.py:1]: removed

**Pull request merged**